Source/Core/DolphinQt2/WiiUpdate.cpp

```diff
@@ -96,7 +96,10 @@ static WiiUtils::UpdateResult ShowProgress(QWidget* parent, Callable function, A
   auto* cancel_button = new QPushButton(QObject::tr("&Cancel"), parent);
   dialog.setCancelButton(cancel_button);
   Common::Flag was_cancelled;
-  QObject::disconnect(&dialog, &QProgressDialog::canceled, &dialog, &QProgressDialog::cancel);
+  // Disconnect the default cancel signal. We don't know if Qt connected it using the old
+  // or the new signal slot syntax, so we need to use the old syntax with a wildcard,
+  // which allows us to get rid of all slots in all cases.
+  QObject::disconnect(&dialog, SIGNAL(canceled()), nullptr, nullptr);
```

Comment:
Okay, I spelunked through [qtbase/src/corelib/kernel/qobject.cpp](https://code.woboq.org/qt5/qtbase/src/corelib/kernel/qobject.cpp.html) and [its header](https://code.woboq.org/qt5/qtbase/src/corelib/kernel/qobject.h.html), and it _looks_ like either way the signal gets converted into its internal QObject signal index and then processed. So the new-style disconnect should work.

Alternatively, we could also probably get away with just disconnecting _everything_: `dialog->disconnect();`.